sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java

```diff
@@ -247,8 +261,103 @@ public Mono<Void> run(CancellationToken cancellationToken) {
+                            }
+
+                            int retryCount = this.streamsConstrainedRetries.incrementAndGet();
+                            boolean shouldRetry = retryCount <= this.maxStreamsConstrainedRetries;
+
+                            if (!shouldRetry) {
+                                logger.error(
+                                    "Lease with token {}: Reached max retries for streams constrained exception with statusCode [{}] : subStatusCode [{}] : message [{}].",
+                                    this.lease.getLeaseToken(),
+                                    clientException.getStatusCode(),
+                                    clientException.getSubStatusCode(),
+                                    clientException.getMessage());
+                                logger.error("Streams constrained retries exhausted.", clientException);
+                                this.resultException = new RuntimeException(clientException);
+                                return Flux.error(throwable);
+                            }
+
+                            logger.warn(
+                                "Lease with token {}: Streams constrained exception encountered, will retry.",
+                                this.lease.getLeaseToken());
+                            logger.warn("Streams constrained retry {}/{}", retryCount, this.maxStreamsConstrainedRetries);
+                            logger.warn("Streams constrained exception.", clientException);
+
+                            if (this.options.getMaxItemCount() <= 1) {
+                                logger.error(
+                                    "Cannot reduce maxItemCount further as it's already at {}",
+                                    this.options.getMaxItemCount());
+                                logger.error("Streams constrained retries exhausted.", clientException);
+                                this.resultException = new RuntimeException(clientException);
+                                return Flux.error(throwable);
+                            }
+
+                            this.options.setMaxItemCount(this.options.getMaxItemCount() / 2);
+                            logger.warn("Reducing maxItemCount, new value: {}", this.options.getMaxItemCount());
+                            return Flux.empty();
+                        }
+                        case PARSING_ERROR:
+
+                            if (!Configs.isChangeFeedProcessorMalformedResponseRecoveryEnabled()) {
+                                logger.error(
+                                    "Lease with token {}: Parsing error encountered. To enable automatic retries, please set the '{}' configuration to 'true'. Failing.",
+                                    this.lease.getLeaseToken(),
+                                    Configs.CHANGE_FEED_PROCESSOR_MALFORMED_RESPONSE_RECOVERY_ENABLED);
+                                logger.error("Parsing error.", clientException);
```

Comment:
same here, let's combine the error log with the partition / lease token logging.

Reply:
Fixed